lib/identifiers/doi.rb

```diff
@@ -1,7 +1,7 @@
 module Identifiers
   class DOI
     def self.extract(str)
-      str.scan(%r{\b10\.\d{3,}/\S+\b}).map(&:downcase)
+      str.scan(%r{\b10.\d{4,9}/[-._;()/:A-Z0-9]+\b}i).map(&:downcase)
```

Comment:
@scottmatthewman  I've sent a quick email to Joe Wass over at CrossRef, to enquire if anyone actually uses that double namespace format. Maybe we can ignore that complication if it's not used.

@mudge I added in the Wiley case, and that'll cover most of those.